spinalcordtoolbox/centerline/core.py

```diff
@@ -236,6 +238,11 @@ def get_centerline(im_seg, param=ParamCenterline(), verbose=1):
         plt.savefig('fig_centerline_' + datetime.now().strftime("%y%m%d-%H%M%S%f") + '_' + param.algo_fitting + '.png')
         plt.close()
 
+    # Save centerline image to tmp_folder (but only if user hasn't opted to `remove_temp_files`)
+    if not remove_temp_files:
+        tmp_folder = tmp_create("centerline")
+        im_centerline.save(os.path.join(tmp_folder, add_suffix(im_seg.absolutepath, "_ctr")), mutable=True)
```

Comment:
Sorry for the confusion! To clarify, there should be 2 temporary folders:

* The direct outputs of the OptiC binary -- these are the `.hdr` and `.img` files
* The output of the `get_centerline()` function -- this should produce a separate `.nii.gz` file (see above)

Could you check to see if a separate temp folder gets created? Perhaps one titled `centerline`?

Reply:
Indeed, two temporary folders are created when running `sct_deepseg_lesion`. However, neither contains `.nii.gz` centerline:

<details><summary> sct_deepseg_lesion  output</summary>

```
$ sct_deepseg_lesion -i sub-mon001_ses-M0_T2w.nii.gz -c t2 -r 0

--
Spinal Cord Toolbox (git-master-812a8e00bd8862c3c9f2d099e24e0143a5337976*)

sct_deepseg_lesion -i sub-mon001_ses-M0_T2w.nii.gz -c t2 -r 0
--


Method:
	Centerline algorithm: svm
	Assumes brain section included in the image: True

Image header specifies datatype 'int16', but array is of type 'float64'. Header metadata will be overwritten to use 'float64'.
Creating temporary folder (/var/folders/v8/8v423b3x4g58mp5g1p33wsh00000gn/T/sct-20221205172432.326449-w7ur5ee2)
Reorient the image to RPI, if necessary...

Finding the spinal cord centerline...
Creating temporary folder (/var/folders/v8/8v423b3x4g58mp5g1p33wsh00000gn/T/sct-20221205172438.717856-hrj4bnha)

Cropping the image around the spinal cord...
Normalizing the intensity...
Image header specifies datatype 'float64', but array is of type 'float32'. Header metadata will be overwritten to use 'float32'.
load data...

Segmenting the MS lesions using deep learning on 3D patches...
Image header specifies datatype 'uint8', but array is of type 'float64'. Header metadata will be overwritten to use 'float64'.
load data...

Reassembling the image...
Resampling the segmentation to the original image resolution...
load data...

Binarizing the segmentation to avoid interpolation effects...

Reorienting the segmentation to the original image orientation...

Done! To view results, run one of the following commands (depending on your preferred viewer):

```
</details>


<details><summary>The first temp folder - contains lesion segmentation</summary>

```
ls -1 /var/folders/v8/8v423b3x4g58mp5g1p33wsh00000gn/T/sct-20221205172432.326449-w7ur5ee2                     (master|…8)
image_in_RPI_res.nii
image_in_RPI_res_norm.nii
image_in_RPI_res_norm_resampled3d.nii
image_in_RPI_res_norm_resampled3d_lesionseg.nii
image_in_RPI_res_norm_resampled3d_lesionseg_resampled2d.nii
```
</details>

<details><summary>The second temp folder - contains centreline .hdr/.img pairs</summary>

```
$ ls -1 /var/folders/v8/8v423b3x4g58mp5g1p33wsh00000gn/T/sct-20221205172438.717856-hrj4bnha                     (master|…8)
img_rpi_uint16.nii
img_rpi_uint16_optic_ctr.hdr
img_rpi_uint16_optic_ctr.img
img_rpi_uint16_optic_ctr.txt
img_rpi_uint16_optic_svm.hdr
img_rpi_uint16_optic_svm.img
```
</details>

Reply:
Oh! I didn't realize there were already 2 temp folders. In that case, what I should have said was: "there should be a third temp folder." :sweat_smile: 

Looking at the folder names

> `sct-20221205172438.717856-hrj4bnha`
>`sct-20221205172432.326449-w7ur5ee2`

This is weird! As per the above lines of code, there should be another temp folder called something like:

> `sct-20221205172432.326449-centerline-w7ur5ee2`

If this folder isn't present, then this PR didn't do what it was intended to do. :sweat:

Reply:
The PR works in the term, that the centreline temporary folder is not deleted when using `-r 0`:

> Finding the spinal cord centerline...
> Creating temporary folder (/var/folders/v8/8v423b3x4g58mp5g1p33wsh00000gn/T/sct-20221205172438.717856-hrj4bnha)

Before the PR, [this temporary folder was always deleted (even using `-r 0`)](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3956#issue-1460413951):

> Finding the spinal cord centerline...
> Creating temporary folder (/var/folders/v8/8v423b3x4g58mp5g1p33wsh00000gn/T/sct-20221122133425.506625-2jy950w_)
> Remove temporary files...
> rm -rf /var/folders/v8/8v423b3x4g58mp5g1p33wsh00000gn/T/sct-20221122133425.506625-2jy950w_

But still, if there should be a third temp folder, it is not preserved even with `-r 0`.